### PR TITLE
Each reuse

### PR DESCRIPTION
**Pull request by mortoray**
Adds the `Each.Reuse` property to allow reusing existing nodes.

This experimental feature has some limtiations:

- https://github.com/fusetools/fuselibs/issues/4218
- https://github.com/fusetools/fuselibs/issues/4217
- https://github.com/fusetools/fuselibs/issues/4219  (More of a feature)

This PR contains:
- [x] Changelog
- [x] Documentation - limited for new API
- [x] Tests

### CODE REVIEW

File: Source/Fuse.Nodes/Visual.Children.uno [hunk 1]
```diff
@@ -187,33 +226,69 @@ namespace Fuse
...
+						OnMoved(c);
+				}
 			}
 			finally
 			{
 				ReleaseRooting(capture);
 			}
 		}
-
+
```
Comment (kusma) [inline Source/Fuse.Nodes/Visual.Children.uno]: please don't. There seems to be some empty indentation-war going on, and it's going to lead to pointless merge-conflicts :P

Comment (mortoray) [inline Source/Fuse.Nodes/Visual.Children.uno]: This is probably an editor setting. Likely somebody is stripping spaces on save. Inside a class it'd be only natural for the indentation to be here, unless you stripped spaces.  Pressing enter will auto-indent the next row.

Do we want to strip trailing spaces?

Comment (kusma) [inline Source/Fuse.Nodes/Visual.Children.uno]: I don't really care so much if it's one way or the other (I have a slight preference for no trailing whitespace, as it just seems pointless), I would prefer if we didn't needlessly change it.

**Comment by kusma** [comment]
This conflicts pretty badly with the current master, would you mind rebasing @mortoray?

**Comment by mortoray** [comment]
@kusma Rebased.

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,12 @@
 - Updated the bundled libjpeg, libpng, Freetype, and SDL2 libaries for macOS to not use AVX instructions, since they are incompatible with the CPUs in some older Mac models. This fixes an issue with SIGILLs in native builds.
 
 ### Native
+## Each Reuse
+- Added `Reuse` to `Each` allowing the reuse of nodes
+- Added `OnChildMoved` to `Visual`. Anything implementing `OnChildAdded` or `OnChildRemoved` will likely need to implement `OnChildMoved` as well. This happens when a child's position in `Children` list changes.
+- Added `OnChildMovedWhileRooted` to `IParentObserver`
+
+## Native
```
Comment (kusma) [inline CHANGELOG.md]: Something's up here.

Comment (kusma) [inline CHANGELOG.md]: I see what happened; this ended up in the wrong section of the changelog.

**Pull request merged**